network-area-diagram-viewer/src/main/resources/svg.js

```diff
@@ -0,0 +1,463 @@
+            // It we were moving a bus node (its circle),
+            // find the id of the corresponding voltage level
+            var id = svgElem.getAttribute("id");
+            var parent = svgElem.parentElement;
+            var idParent = parent.getAttribute("id");
+            if (idParent) {
+                id = idParent;
+            }
+            updateOnClient(id, position, translation);
+        }
+    }
+
+    function updateOnClient(id, position, translation) {
+        console.log("updateOnClient, moved node id = " + id);
+        translateText(id, translation);
+        // TODO(Luma) Update here the position of all inner busNodes
+        for (var edge of svg.getElementsByTagName("nad:edge")) {
+            var node1 = edge.getAttribute("node1");
+            var node2 = edge.getAttribute("node2");
+            console.log("  consider edge (" + node1 + ", " + node2 + ") after moved node " + id);
+            if (node1 === id || node2 === id) {
+                updateEdge(edge, node1, node2, id, position, translation);
+            }
+        }
+    }
+
+    function updateEdge(edgeMetadata, node1, node2, movedNodeId, position, translation) {
+        const DEBUG_EXPERIMENTS = false;
+        if (DEBUG_EXPERIMENTS) {
+            updateEdgeExperiments(edgeMetadata, node1, node2, movedNodeId, position, translation);
+        } else {
+            updateEdgeUsingRotation(edgeMetadata, node1, node2, movedNodeId, position, translation);
+        }
+    }
+
+    function updateEdgeUsingRotation(edgeMetadata, node1, node2, movedNodeId, position, translation) {
+        // This edge is adjacent to the moved node
+        console.log("  adjacent edge (" + node1 + ", " + node2 + ")");
+        var edgeId = edgeMetadata.getAttribute("svgid");
+        var edgeSvg = svg.getElementById(edgeId);
+        if (!edgeSvg) {
+            return;
+        }
+
+        // Translate all edge-related graphical elements to the new position
+        translate(svg, edgeSvg, translation);
+
+        // If the other node of the edge is visible in the diagram,
+        // rotate the edge towards it.
+        // rotation centered on moved node
+        // (now we may not "reach" the other node, but we point in the right direction)
+        var otherNodeId = movedNodeId === node1 ? node2 : node1;
+        var otherNodeSvg = svg.getElementById(otherNodeId);
+        console.log("  other node is " + otherNodeId + ", svg = " + otherNodeSvg);
+        if (otherNodeSvg) {
+            rotateEdge(edgeId, edgeSvg, movedNodeId, otherNodeId, otherNodeSvg, position);
+        }
+    }
+
+    function rotateEdge(edgeId, edgeSvg, movedNodeId, otherNodeId, otherNodeSvg, position1) {
+        var otherNodeCenter = center(otherNodeSvg);
+        var movedNodeMetadata = svg.querySelectorAll('nad\\\\:node[svgid="' + movedNodeId + '"]')[0];
+        var otherNodeMetadata = svg.querySelectorAll('nad\\\\:node[svgid="' + otherNodeId + '"]')[0];
```

Comment:
As you want only the first, you'd rather use querySelector to be more efficient.

Reply:
Great, thanks for the hint!

Reply:
done